src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/Semantics/ConstVal.cs

```diff
@@ -199,5 +128,51 @@ private bool IsDefault(object o)
 
             return false;
         }
+
+        public static ConstVal GetDefaultValue(ConstValKind kind)
+        {
+            switch (kind)
+            {
+                case ConstValKind.Int:
+                    return Zero;
+
+                case ConstValKind.Double:
+                    return new ConstVal(0.0);
+
+                case ConstValKind.Long:
+                    return new ConstVal(0L);
+
+                case ConstValKind.Decimal:
+                    return new ConstVal(0M);
+
+                case ConstValKind.Float:
+                    return new ConstVal(0F);
```

Comment:
Are these other defaults (long, double, decimal, float) not used enough to warrant having their own cached static instances?